org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JDTUtils.java

```diff
@@ -1182,15 +1183,39 @@ public static IResource findResource(URI uri, Function<URI, IResource[]> resourc
 							JavaLanguageServerPlugin.logException(e.getMessage(), e);
 					}
 				}
+
+				//find the resource which has PackageFragmentRoot
 				//find closest project containing that file, in case of nested projects

```

Comment:
```suggestion
				//find closest project containing that file, in case of nested projects
				//ignore files outside a source folder (https://github.com/eclipse-jdtls/eclipse.jdt.ls/issues/3447)
```

Reply:
However, the method name `findResource` no longer matches the contract now. This should be addressed.